### PR TITLE
Remove code that got reintroduced during a merge.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs [hunk 1]
```diff
@@ -576,7 +576,8 @@ private static bool IsAllDigits(string filterText)
                 }
 
                 // Switch to soft selection, if user moved caret to the start of a non-empty filter span.
-                // This prevents commiting if user types a commit character at this position later, but still has the list if user types filter character
+                // This prevents commiting if user types a commit character at this position later, but 
+                // still has the list if user types filter character
                 // i.e. blah| -> |blah -> !|blah
                 // We want the filter span non-empty because we still want hard selection in the following case:
                 //
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs]: the code below this is where the code in Controller_CaretPositionChanged.cs moved to.  So we don't need that other code anymore.

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_CaretPositionChanged.cs [hunk 2]
```diff
@@ -42,21 +42,6 @@ internal override void OnCaretPositionChanged(object sender, EventArgs args)
                 dismissIfEmptyAllowed: false,
                 recheckCaretPosition: true,
                 filterState: null);
-
-            if (model.SelectedItemOpt != null && model.IsHardSelection)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_CaretPositionChanged.cs]: I moved this code, and ravi changed the code.  We don't need this code anymore.

**Comment by CyrusNajmabadi** [comment]
Tagging @DustinCampbell @dotnet/roslyn-ide @MattGertz

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_CaretPositionChanged.cs [hunk 3]
```diff
@@ -42,21 +42,6 @@ internal override void OnCaretPositionChanged(object sender, EventArgs args)
...
-                // We want the filter span non-empty because we still want completion in the following case:
-                // A a = new | -> A a = new (|
-
-                var currentSpan = model.GetViewBufferSpan(model.SelectedItemOpt.Span).TextSpan;
-                if (caretPoint == currentSpan.Start && currentSpan.Length > 0)
-                {
-                    sessionOpt.SetModelIsHardSelection(false);
-                }
-            }
         }
```
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_CaretPositionChanged.cs]: Thanks! This is exactly the change I wasn't sure about.

**Comment by DustinCampbell** [comment]
Note: We shouldn't need approval to get this in, since it's just fixing up the regular merge from dev15-rc2 I did earlier: https://github.com/dotnet/roslyn/pull/15627. Once tests pass, we can just merge this.

**Comment by CyrusNajmabadi** [comment]
Great!

**Pull request merged**